### ISSUE TITLE
Failed to build pg gem - cannot load such file mkmf

### ISSUE THREAD

**Issue by themasterchef** [hw-cookbooks/postgresql#165]
When I run the Postgresql server or ruby recipes on the prepackaged ubuntu/trusty64 Vagrant box from vagrant cloud, I get the following error:

```
Error executing action `install` on resource 'chef_gem[pg]'
===================


Gem::Installer::ExtensionBuildError
-----------------------------------
ERROR: Failed to build gem native extension.

        /usr/bin/ruby1.9.1 extconf.rb
/usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require': cannot load such file -- mkmf (LoadError)
	from /usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require'
	from extconf.rb:2:in `<main>'


Gem files will remain installed in /var/lib/gems/1.9.1/gems/pg-0.17.1 for inspection.
Results logged to /var/lib/gems/1.9.1/gems/pg-0.17.1/ext/gem_make.out



Cookbook Trace:
---------------
/tmp/vagrant-chef-3/chef-solo-1/cookbooks/postgresql/recipes/ruby.rb:58:in `rescue in from_file'
/tmp/vagrant-chef-3/chef-solo-1/cookbooks/postgresql/recipes/ruby.rb:24:in `from_file'


Resource Declaration:
---------------------
# In /tmp/vagrant-chef-3/chef-solo-1/cookbooks/postgresql/recipes/ruby.rb

 58:     chef_gem "pg"
 59:   rescue Gem::Installer::ExtensionBuildError => e



Compiled Resource:
------------------
# Declared in /tmp/vagrant-chef-3/chef-solo-1/cookbooks/postgresql/recipes/ruby.rb:58:in `rescue in from_file'

chef_gem("pg") do
  provider Chef::Provider::Package::Rubygems
  action :install
  retries 0
  retry_delay 2
  package_name "pg"
  version "0.17.1"
  cookbook_name :postgresql
  recipe_name "ruby"
end
```

One possible solution is described in http://blogs.law.harvard.edu/hoanga/2006/10/27/fixing-mkmf-load-error-ruby-in-ubuntu/. These are the important bits:

"If you’re a Debian or Ubuntu user you’ll find that the Ruby standard distribution is split into lots of little packages so doing something like apt-get install ruby only gives you the ruby binary and a subset of the libraries for Ruby. You’ll need to add more packages if you want to utilize more of Ruby’s standard library. This is all in order to comply with Debian packaging Guidelines."

...and...

"To get around [the mkmf error] you will need to install ruby1.8-dev (or ruby1.6-dev or whatever version of ruby you’re running) to get mkmf.rb which is listed in the Ruby Standard Library."

In other words, putting a `sudo apt-get install ruby1.[version]-dev` in the Chef recipe might fix this bug.

Update: when running Chef on Ubuntu trusty, ruby 1.9.3 is what gets installed by default, but there is no ruby1.9.3-dev package in apt. Instead, you will actually need to obtain its alias, ruby1.9.1-dev, which http://askubuntu.com/questions/284459/couldnt-find-any-package-by-regex-ruby says will do what is needed.

Update: since the pg gem building happens *before* the convergence stage, this package must be installed at the recipe compilation stage or earlier.

**Comment by slyness** [comment]
building pg gem is not longer a problem.
